src/Phinx/Db/Plan/Plan.php

```diff
@@ -442,61 +405,62 @@ protected function gatherCreates($actions)
      */
     protected function gatherUpdates($actions)
     {
-        collection($actions)
-            ->filter(function ($action) {
-                return $action instanceof AddColumn
+        foreach ($actions as $action) {
+            if (
+                !(
+                    $action instanceof AddColumn
                     || $action instanceof ChangeColumn
                     || $action instanceof RemoveColumn
-                    || $action instanceof RenameColumn;
-            })
-            // We are only concerned with table changes
-            ->reject(function ($action) {
-                return isset($this->tableCreates[$action->getTable()->getName()]);
-            })
-            ->each(function ($action) {
-                $table = $action->getTable();
-                $name = $table->getName();
+                    || $action instanceof RenameColumn
+                )
+                || isset($this->tableCreates[$action->getTable()->getName()])
+            ) {
```

Comment:
Let's just break this up into two if statements so it's cleaner.

Reply:
done